test/authController.test.js

```diff
@@ -0,0 +1,108 @@
+const request = require('supertest');
+const { intervalId: orderUpdateInterval } = require('../src/models/Order');
+
+// Declare variables for the server, database connection, and in-memory MongoDB instance
+let server;
+let mongooseConnection;
+let mongodb;
+
+// Credentials for test user
+const testUserCredentials = {
+  email: 'ava@ava.com',
+  password: 'secret',
+};
+
+// Test user data
+const testUser = {
+  name: 'ava',
+  username: 'ava',
+  email: 'ava@ava.com',
+  password: 'secret',
+  role: 'user',
+};
+
+// log in and return a cookie
+const loginAndReturnCookie = async (credentials) => {
+  // Send a login request and get the signed cookie from the response
+  const resp = await request(app).post('/api/v1/auth/login').send(credentials);
+
+  // Get the 'set-cookie' header from the response
+  const cookieHeader = resp.headers['set-cookie'];
+  // Find the cookie that starts with 'token' in the 'set-cookie' header
+  // Split the cookie string by ';' and take the first part (before the first ';')
+  const signedCookie = cookieHeader
+    .find((cookie) => cookie.startsWith('token'))
+    .split(';')[0];
+  return signedCookie; // Return the signed cookie, which typically contains authentication token
+};
+
+// set up the mongodb and the express server before starting the tests
+beforeAll(async () => {
+  // This will create a new instance of "MongoMemoryServer" and automatically start it
+  mongodb = await MongoMemoryServer.create();
+  const url = mongodb.getUri();
+  // set the url so that our server's mongoose connects to the in-memory mongodb and not our real one
+  process.env.MONGO_URL = url;
+  mongooseConnection = await connectDB(url);
+  server = await app.listen(8001);
+});
+
+afterAll(async () => {
+  // turn off the server and mongo connections once all the tests are done
+  await server.close();
+  await mongooseConnection.disconnect();
+  await mongodb.stop();
+  // turn off the order update interval so that jest can cleanly shutdown
+  clearInterval(orderUpdateInterval);
+});
+
+describe('Authentication API Endpoints', () => {
+  it('should register a new user and log in', async () => {
+    // Arrange: Register a new user
+    const registrationResponse = await request(app)
+      .post('/api/v1/auth/register')
+      .send(testUser);
+
+    // Assert: Check the response status and body
+    expect(registrationResponse.status).toBe(201); // Expecting a successful registration
+    expect(registrationResponse.body).toHaveProperty('user'); // Expecting a user object in the response
+
+    // Act: Log in with the newly registered user's credentials
+    const loginResponse = await request(app)
+      .post('/api/v1/auth/login')
+      .send(testUserCredentials);
+
+    // Assert: Check the response status and body
+    expect(loginResponse.status).toBe(201); // Expecting a successful login
+    expect(loginResponse.body).toHaveProperty('user'); // Expecting a user object in the response
```

Comment:
Here we could also have further assertions on the shape of the object inside `loginResponse.body.user`

Reply:
<img width="728" alt="jest1" src="https://github.com/Code-the-Dream-School/dd-prac-team4-back/assets/95896159/a6b1f4c7-2920-4db0-90bb-07284db5a3ed">
With each unit test, it becomes clearer and more familiar. Thank you very much!❤️